srv/components/system/config/hosts.sls

```diff
@@ -16,26 +16,22 @@
 #
 
 hostsfile:
-  file.managed:
-    - name: /etc/hosts
-    - contents: |
-        {%- if pillar['cluster']['node_list']|length > 1 %}
-        127.0.0.1     localhost localhost.localdomain localhost4 localhost4.localdomain4
-        ::1           localhost localhost.localdomain localhost6 localhost6.localdomain6
-        -------------------------------------------------------------------------------
-        {%- for node in pillar['cluster']['node_list'] %}
-        {%- if pillar['cluster'][node]['network']['data_nw']['pvt_ip_addr'] %}
-        {{ pillar['cluster'][node]['network']['data_nw']['pvt_ip_addr'] }}   {{ node -}}
-        {%- else %}
-        {%- for srvnode, ip_data in salt['mine.get'](node, 'node_ip_addrs') | dictsort() %}
-        {{ ip_data[pillar['cluster'][srvnode]['network']['data_nw']['iface'][1]][0] }}   {{ srvnode -}}
-        {% endfor -%}
-        {% endif -%}
-        {% endfor %}
-        {%- else %}
-        127.0.0.1     localhost localhost.localdomain localhost4 localhost4.localdomain4 {{ grains['id'] }}
-        ::1           localhost localhost.localdomain localhost6 localhost6.localdomain6
-        -------------------------------------------------------------------------------
-        {%- endif %}
-    - user: root
-    - group: root
+ file.blockreplace:
+ - name: /etc/hosts
+ - backup: False
+ - marker_start: "#---pvt_data_start---"
```

Comment:
For unfamiliar reviewers:
These markers ensure that the changes are made only within this block.
If block is not found in the file, it would be appended to the end of file along with the markers.

Ref:
https://docs.saltstack.com/en/3001/ref/states/all/salt.states.file.html#salt.states.file.blockreplace